src/main/java/com/ivan/salesapp/web/controllers/CartController.java

```diff
@@ -0,0 +1,145 @@
+        this.iProductService = iProductService;
+        this.iUserService = iUserService;
+        this.iOrderService = iOrderService;
+        this.modelMapper = modelMapper;
+    }
+
+    @PostMapping("/add-product")
+    @PreAuthorize("isAuthenticated()")
+    public ModelAndView addToCartConfirm(String id, int quantity, HttpSession session) {
+        ProductDetailsViewModel product = this.modelMapper
+                .map(this.iProductService.findProductById(id), ProductDetailsViewModel.class);
+
+        OrderProductViewModel orderProductViewModel = new OrderProductViewModel();
+        orderProductViewModel.setProduct(product);
+        orderProductViewModel.setPrice(product.getPrice());
+
+        ShoppingCartItem cartItem = new ShoppingCartItem();
+        cartItem.setProduct(orderProductViewModel);
+        cartItem.setQuantity(quantity);
+
+        List<ShoppingCartItem> cart = this.retrieveCart(session);
+        this.addItemToCart(cartItem, cart);
+
+        return super.redirect("/home");
+    }
+
+    @GetMapping("/details")
+    @PreAuthorize("isAuthenticated()")
+    public ModelAndView cartDetails(ModelAndView modelAndView, HttpSession session) {
+        var cart = this.retrieveCart(session);
+        modelAndView.addObject("totalPrice", this.calcTotal(cart));
+
+        return super.view("cart/cart-details", modelAndView);
+    }
+
+    @DeleteMapping("/remove-product")
+    @PreAuthorize("isAuthenticated()")
+    public ModelAndView removeFromCartConfirm(String id, HttpSession session) {
+        this.removeItemFromCart(id, this.retrieveCart(session));
+
+        return super.redirect("/cart/details");
+    }
+
+    @PostMapping("/checkout")
+    @PreAuthorize("isAuthenticated()")
+    public ModelAndView checkoutConfirm(HttpSession session, Principal principal) {
+        var cart = this.retrieveCart(session);
+
+        OrderServiceModel orderServiceModel = this.prepareOrder(cart, principal.getName());
+        this.iOrderService.createOrder(orderServiceModel);
+        return super.redirect("/home");
+    }
+
+    private List<ShoppingCartItem> retrieveCart(HttpSession session) {
+        this.initCart(session);
+
+        return (List<ShoppingCartItem>) session.getAttribute("shopping-cart");
+    }
+
+    private void initCart(HttpSession session) {
+        if (session.getAttribute("shopping-cart") == null) {
+            session.setAttribute("shopping-cart", new LinkedList<>());
+        }
+    }
+
+    private void addItemToCart(ShoppingCartItem item, List<ShoppingCartItem> cart) {
+        for (ShoppingCartItem shoppingCartItem : cart) {
+            if (shoppingCartItem.getProduct().getProduct().getId().equals(item.getProduct().getProduct().getId())) {
+                shoppingCartItem.setQuantity(shoppingCartItem.getQuantity() + item.getQuantity());
+                return;
+            }
```

Comment:
remove return and just add else statement 
if(..){
                shoppingCartItem.setQuantity(shoppingCartItem.getQuantity() + item.getQuantity());
}else{
        cart.add(item);

}